### ISSUE TITLE
MacOS Sonoma problems

### ISSUE THREAD

**Issue by Fred-DTV** [dado3212/spaces-renamer#94]
Tried through the 1.2.0 official MacForge release as well as through the 1.2.2 beta both times with re-installing space renamer.
It shows up in the taskbar and I can use the pop up window to enter names for the spaces (it even recognizes the correct number of spaces) but it doesn't rename the spaces.
SIP and everything else is turned off.
 
```
Fred@FredBook-Pro ~ % ps aux | grep -v grep | grep -i macforge
Fred              1077   0.0  0.1 410095232  37024   ??  S    Sun10PM   0:03.64 /Applications/MacForge.app/Contents/Library/LoginItems/MacForgeHelper.app/Contents/MacOS/MacForgeHelper
Fred@FredBook-Pro ~ % ps aux | grep -v grep | grep -i spacesrenamer
Fred              1071   0.0  0.0 35764792  13000   ??  S    Sun10PM   0:01.86 /Applications/SpacesRenamer.app/Contents/MacOS/SpacesRenamer
Fred@FredBook-Pro ~ % csrutil status
System Integrity Protection status: disabled.
```

**Comment by wlj-iii** [comment]
Looking through at a very passing glance (on my way to uninstall), there is no /Library/Containers directory, potentially moved in Sonoma along with everything else :(

**Comment by jcallahan526** [comment]
Same here.   Was working brilliantly with previous MacOS

**Comment by izaankml** [comment]
Same issue where spaces names do not update, although spaces renamer is always identifying one more space than I currently have open also

**Comment by irishtoon** [comment]
Won't install for me at all on Sonoma 14.3, any advice? 

SpaceRenamers.app can't be opened because Apple cannot check it for malicious software. This software needs to updated. Contact the developer for more information.

**Comment by Itzik217** [comment]
Any updates here? i'm getting the same thing. Followed all steps and in my case i don't even haev the macenhance folder.

**Comment by trentbrew** [comment]
Same issue here ever since upgrading to Sonoma 14.4

**Comment by dado3212** [comment]
Sonoma 14.4 is indeed broken, but previous versions through 14.3 have been validated. Tracking the 14.4 issue in #95.

**Comment by dennisrcao** [comment]
Anyone have a working spaces renamer on their Sonoma M1? I just updated and this stopped working.

**Comment by dado3212** [comment]
I updated the readme - https://github.com/dado3212/spaces-renamer?tab=readme-ov-file#installation-m1m2apple-silicon.

You can try using the version that has MacForge 1.2.2 (3).

**Comment by nynewco** [comment]
I tried all the steps above. M1 Sonoma 14.5. Didn't work.

Steps I took:

Recovery: csrutil disable
Installed MacForge. 
Ensured it is 1.2.2 (3) even after relaunching
Recovery: csrutil enable --without debug --without fs --without nvram --without kext
Restarted Machine
Copied spaces-renamer bundle to Plugin folder
Killed the dock
Could not see Spaces Renamer icon in menu bar on top
Double clicked Spaces renamer -- first time, it starts the launching animation, but nothing happens. Second time, it kind of highlights the window below it, and nothing happens. I cannot see the Spaces Renamer icon in the menu bar.

Some observations. DockKit shows red in MacForge plugin 
![image](https://github.com/dado3212/spaces-renamer/assets/87440000/9d0530b4-0bc7-46f4-864f-10ac3be11843)

When I uninstalled Spaces Renamer before, there were 4 locations it existed. 
![image](https://github.com/dado3212/spaces-renamer/assets/87440000/df28412f-ee22-4338-bdaf-372f6f7f4b9d)

Now if I try to uninstall it, I only see 3 locations. Don't see the Library/Preferences/com.alexbeals.spacesrenamer.plist

![image](https://github.com/dado3212/spaces-renamer/assets/87440000/e2c531de-728b-44dc-a4fd-6dff5d867bea)

**Comment by dennisrcao** [comment]
It works for me(Sonoma 14.4.1) however the only thing is when using multiple desktops all the other desktops have the tray icon on the top right corner, but my MBP itself does not. 

![SCR-20240521-llyy](https://github.com/dado3212/spaces-renamer/assets/521934/7ee01fb2-1711-4290-a848-d4a65f00fdb6)
^ This icon shows for my other monitors. however is not present on my MBP.

**Comment by jvthomas90** [comment]
@dennisrcao Based off of the screenshot you shared in #91 I'm assuming you actually mean "multiple monitors" 

> rather than 
> 
> - "desktops" (to use Windows' language) 
> - or "desks" (ChromeOS spinoff) 
> - or "workspaces" (Linux taxonomy) 
> - or "spaces" (MacOS lingo) 
> 
> etc etc i.e. the virtual version of having multiple screens worth of app/window groupings to work with.

If my assumption is right and you're referring to the visibility of the SpacesRenamer icon on monitors 2 and 3 as opposed to the lack of visibility on the built-in MBP screen
- then the good news is that Spaces Renamer is indeed running _(you can confirm this by opening the native Activity Monitor app and searching for Spaces Renamer among the list of currently active processes)_ 
- though the problem seems to be that the resolution of your main display is so low that there isn't enough space to see some of the status-bar icons that get shoveled off all the way to the left of your screen _(either ending up behind the notch, if your MBP has one, or the active app's menu-items that are on the far left of the menu-bar)_

You can follow this tip (https://github.com/dado3212/spaces-renamer/issues/95#issuecomment-2116313964) for a quick and dirty natively supported fix to this otherwise minor visual glitch, or alternatively you can utilize the 3rd party app I suggested in #91 for a better-suited long-term solutions e.g. splitting up your menu-bar items by either a dedicated sub-group of apps (think "folders, for the menu bar") or the more generic "bartender bar" (basically, a second menu-bar). That's for easy visual/mouse pointer access, mind you, the two tips regarding keyboard access that I shared on #91 apply too.

TL;DR version 
1. you can either shrink your MBP screen temporarily to be able to see the icon then CTRL+CLICK it to drag it farther to the right-hand side so that when you "zoom-in" again _(i.e. dial-down the resolution back to the default or your preferred value)_ it still remains visible on the main display. 
<img width="827" alt="Screenshot 2024-05-23 at 06 02 26" src="https://github.com/dado3212/spaces-renamer/assets/69699907/de22373b-d71c-43b5-aa4d-af4bffeb9aa0">

2. Or you can install Bartender and take the time to configure it so that any and all status-bar icons are better organized, such that they are either visible, or if not visible then they still remain easily accessible by various mouse pointer or keyboard tricks.
<img width="1013" alt="Screenshot 2024-05-23 at 06 03 05" src="https://github.com/dado3212/spaces-renamer/assets/69699907/ad31b606-7aa1-4a54-b297-d2f9e806dae4">

**Comment by dennisrcao** [comment]
@jvthomas90 Correct - I meant "multiple monitors" apologies. And thank you it looks like the resolution trick #1  worked.

**Comment by dantab66** [comment]
Hi, I managed to install on a Macbook Pro M3  with Sonoma 14.5 using the new instructions in the FAQ on using the Beta MacForge. 

The first time that Spaces Renamer ran, I was able to set the name of the Desktops which was great.

I now have rebooted, and when I go to set the name of a Desktop it doesn't work and the name stays the same (Desktop 1, Desktop 2, etc..).

I verified that the Spaces Renamer is running.

Any idea why it worked the first time, and now no longer works?

**Comment by dado3212** [comment]
I would make sure that MacForge is also running. Post reboot it may not always start. Trigger it, and then run `pkill -9 Dock` and see if that works.

**Comment by dantab66** [comment]
Thanks for the suggestion.

I did as you said, and it worked!

Strange though. When I did the “pkill” command I received a message that made me believe that it didn’t work.

 "pkill: signalling pid 531: Operation not permitted”

Is there anything I can do to make sure that it comes up properly after reboot?

-Dan

**Comment by dado3212** [comment]
Open System Settings > Login Items and delete MacForgeHelper. Then open MacForge again. It should re-add it as a Login Item, and hopefully stay working again.

**Comment by Gsairus** [comment]
> Hi, I managed to install on a Macbook Pro M3 with Sonoma 14.5 using the new instructions in the FAQ on using the Beta MacForge.
> 
> The first time that Spaces Renamer ran, I was able to set the name of the Desktops which was great.
> 
> I now have rebooted, and when I go to set the name of a Desktop I get the pop up to set the name and click "Update Names" it doesn't seem to do anything anymore. i.e. The name stays the same (Desktop 1, Desktop 2, etc..).
> 
> I verified that the Spaces Renamer is running.
> 
> Any idea why it worked the first time, and now no longer works?

I have the same problem. I was so happy it did finally work after monthes of trying. Now after restart I stopped work. Forge lunches, I reinstaller ForgHelper and tried anything what is described here and not only. I really hope with a new version it will work with a new MacOS. Is it anyway to rewrite the script which will work without MacForge? This App is actually brilliant, it is where Apple screw up with th idea how to design Spaces in the right way.

**Comment by ursprung66** [comment]
Hi, I am on Tahoe 26.1 (25B78) and would love to use this app. Are there any plans to update/redesign? I desperately need such an app! Thanks!
